Platform/Plugins/com.tle.platform.common/src/com/tle/common/util/ExecUtils.java

```diff
@@ -52,6 +54,104 @@
     "", ".exe", ".bat"
   }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
 
+  /**
+   * For a given Process ID, kill any child processes and then kill the process. Works on Unix only,
+   * as it leverages pgrep and kill commands.
+   *
+   * @param pid The Process for which to terminate including it's direct children.
+   */
+  public static void killLinuxProcessTree(int pid) {
+    if (pid > 0) {
+      // get child process PIDs as strings
+      String[] children = getChildUnixProcessPids(pid);
+      // Kill child process(es)
+      for (String child : children) {
+        sendSigKill(Integer.parseInt(child));
+      }
+      // kill process itself
+      sendSigKill(pid);
+    }
+  }
+
+  /**
+   * Runs pgrep -P for a given Process ID, to get a list of child processes as Process IDs.
+   *
+   * @param pid The parent process ID to check for child processes.
+   * @return String[] An array of process IDs for the children of pid. Can be empty.
+   */
+  public static String[] getChildUnixProcessPids(int pid) {
+    try {
+      Process getChildPid = Runtime.getRuntime().exec("pgrep -P " + pid);
+      getChildPid.waitFor();
+      StringBuilder childPid = new StringBuilder();
+      if (getChildPid.exitValue() == 0) {
+        CharStreams.copy(new InputStreamReader(getChildPid.getInputStream()), childPid);
+      } else {
+        StringBuilder errorOutput = new StringBuilder();
+        CharStreams.copy(new InputStreamReader(getChildPid.getErrorStream()), errorOutput);
+        LOGGER.debug("getChildPid function did not run properly.\n" + errorOutput);
+      }
+      getChildPid.destroy();
+      return childPid.toString().replaceAll("\n", " ").split(" ");
+    } catch (IOException | InterruptedException e) {
+      LOGGER.error("Error getting child processes for: " + pid, e);
+    }
+    return new String[] {};
+  }
+
+  /**
+   * Creates a process which then sends a SIGKILL signal to a given process.
+   *
+   * @param pid the Process ID of the process to kill.
+   * @return the exitValue of the SIGKILL process (not the process being killed)
+   */
+  public static int sendSigKill(int pid) {
```

Comment:
Seeing `kill` has a definition of:

```
kill [options] <pid> [...]
```

It'd make sense to allow this to support a list of PIDs too:

```suggestion
  public static int sendSigKill(int[] pid) {
```

That also then supports your use case of killing potentially multiple PIDs. If you did also want to support the form of taking a single PID (which I'm not sure you have a use for here), then you could add an overload that does the transform simply to an array and chain to this method.